bundles/org.openhab.binding.neohub/src/main/java/org/openhab/binding/neohub/internal/NeoBaseHandler.java

```diff
@@ -0,0 +1,231 @@
+import org.eclipse.smarthome.core.types.Command;
+import org.eclipse.smarthome.core.types.RefreshType;
+import org.eclipse.smarthome.core.types.State;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * The {@link NeoBaseHandler} is the OpenHab Handler for NeoPlug devices
+ * 
+ * @author Andrew Fiddian-Green - Initial contribution
+ * 
+ */
+public class NeoBaseHandler extends BaseThingHandler {
+
+    protected final Logger logger = LoggerFactory.getLogger(NeoBaseHandler.class);
+
+    protected NeoBaseConfiguration config;
+
+    /*
+     * an object used to de-bounce state changes OpenHab <=> NeoHub
+     */
+    protected NeoHubDebouncer debouncer = new NeoHubDebouncer();
+
+    public NeoBaseHandler(Thing thing) {
+        super(thing);
+    }
+
+    // ======== BaseThingHandler methods that are overridden =============
+
+    /*
+     * overridden method of BaseThingHandler: by which OpenHab issues commands to
+     * the handler
+     */
+    @Override
+    public void handleCommand(ChannelUID channelUID, Command command) {
+        if (command == RefreshType.REFRESH) {
+            @Nullable
+            NeoHubHandler hub;
+
+            if ((hub = getNeoHub()) != null) {
+                hub.startFastPollingBurst();
+            }
+            return;
+        }
+
+        toNeoHubSendCommandSet(channelUID.getId(), command);
+    }
+
+    /*
+     * overridden method of BaseThingHandler: by which OpenHab initializes the
+     * handler
+     */
+    @Override
+    public void initialize() {
+        config = getConfigAs(NeoBaseConfiguration.class);
+
+        if (config == null || config.deviceNameInHub == null || config.deviceNameInHub.isEmpty()) {
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.CONFIGURATION_ERROR,
+                    String.format("configuration error for %s, status => offline!", getThing().getLabel()));
+            return;
+        }
+
+        updateStatus(ThingStatus.ONLINE, ThingStatusDetail.NONE,
+                String.format("%s initialized, status => online..", getThing().getLabel()));
+    }
+
+    // ======== helper methods used by this class or descendants ===========
+
+    /*
+     * this method is called back by the NeoHub handler to inform this handler about
+     * polling results from the hub handler
+     */
+    public void toBaseSendPollResponse(NeoHubInfoResponse pollResponse) {
+        NeoHubInfoResponse.DeviceInfo myPollResponse = pollResponse.getDeviceInfo(config.deviceNameInHub);
+
+        if (myPollResponse == null) {
+            if (getThing().getStatus() == ThingStatus.ONLINE) {
+                updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.CONFIGURATION_ERROR,
+                        String.format("hub has no info for %s, status => offline!", getThing().getLabel()));
+            }
+            return;
+        }
+
+        if (myPollResponse.isOffline()) {
+            if (getThing().getStatus() == ThingStatus.ONLINE) {
```

Comment:
remove the if, we can always update to OFFLINE

Reply:
Ok, done.